Scripts/DowloadPortalItemData.swift

```diff
@@ -245,31 +246,46 @@ var downloadedItems = previousDownloadedItems
 let dispatchGroup = DispatchGroup()
 
 portalItems.forEach { portalItem in
+    // Checks to see if an item is already downloaded.
+    guard !downloadedItems.keys.contains(portalItem.identifier) else {
+        print("note: Item already downloaded: \(portalItem.identifier)")
+        return
```

Comment:
Two suggestions in one (since they are intertwined):

1. I don't know how efficient `Dictionary.Keys.contains(_:)` is. Better to just subscript into the dictionary.
2. `forEach(_:)` is intended for use at the end of a method chain; a for loop should have been used instead.

```suggestion
for portalItem in portalItems {
    // Checks to see if an item is already downloaded.
    guard downloadedItems[portalItem.identifier] == nil else {
        print("note: Item already downloaded: \(portalItem.identifier)")
        continue
```

Reply:
`contains` should have [asymptotic O(1) performance](https://forums.swift.org/t/does-dictionary-keys-contains-have-o-1-time-complexity/21647/14), though it is not clearly documented by Apple. I initially suggested this change to use contains, but I'm OK just use subscript.

Reply:
Thanks! I actually went and looked at the source, but it wasn't clear from that either. Since `keys` is an instance of a custom type (`Keys`), it makes sense that it could be implemented efficiently for hashable key types.